src/components/Settings/SettingsNetwork/index.tsx

```diff
@@ -0,0 +1,461 @@
+import Button from '@app/components/Common/Button';
+import LoadingSpinner from '@app/components/Common/LoadingSpinner';
+import PageTitle from '@app/components/Common/PageTitle';
+import Tooltip from '@app/components/Common/Tooltip';
+import SettingsBadge from '@app/components/Settings/SettingsBadge';
+import globalMessages from '@app/i18n/globalMessages';
+import defineMessages from '@app/utils/defineMessages';
+import { ArrowDownOnSquareIcon } from '@heroicons/react/24/outline';
+import type { MainSettings } from '@server/lib/settings';
+import { Field, Form, Formik } from 'formik';
+import { useIntl } from 'react-intl';
+import { useToasts } from 'react-toast-notifications';
+import useSWR, { mutate } from 'swr';
+import * as Yup from 'yup';
+
+const messages = defineMessages('components.Settings.SettingsNetwork', {
+  toastSettingsSuccess: 'Settings saved successfully!',
+  toastSettingsFailure: 'Something went wrong while saving settings.',
+  network: 'Network',
+  networksettings: 'Network Settings',
+  networksettingsDescription:
+    'Configure network settings for your Jellyseerr instance.',
+  csrfProtection: 'Enable CSRF Protection',
+  csrfProtectionTip: 'Set external API access to read-only (requires HTTPS)',
+  csrfProtectionHoverTip:
+    'Do NOT enable this setting unless you understand what you are doing!',
+  trustProxy: 'Enable Proxy Support',
+  trustProxyTip:
+    'Allow Jellyseerr to correctly register client IP addresses behind a proxy',
+  forceIpv4First: 'IPv4 Resolution First',
+  forceIpv4FirstTip:
+    'Force Jellyseerr to resolve IPv4 addresses first instead of IPv6',
+  dnsServers: 'Custom DNS Servers',
+  dnsServersTip:
+    'Comma-separated list of custom DNS servers, e.g. "1.1.1.1,[2606:4700:4700::1111]"',
+  proxyEnabled: 'HTTP(S) Proxy',
+  proxyHostname: 'Proxy Hostname',
+  proxyPort: 'Proxy Port',
+  proxySsl: 'Use SSL For Proxy',
+  proxyUser: 'Proxy Username',
+  proxyPassword: 'Proxy Password',
+  proxyBypassFilter: 'Proxy Ignored Addresses',
+  proxyBypassFilterTip:
+    "Use ',' as a separator, and '*.' as a wildcard for subdomains",
+  proxyBypassLocalAddresses: 'Bypass Proxy for Local Addresses',
+  validationProxyPort: 'You must provide a valid port',
+});
+
+const SettingsMain = () => {
+  const { addToast } = useToasts();
+  const intl = useIntl();
+  const {
+    data,
+    error,
+    mutate: revalidate,
+  } = useSWR<MainSettings>('/api/v1/settings/main');
```

Comment:
Shouldn't we also migrate the settings to a dedicated network block and also backend route to call /settings/network instead of main?

Reply:
I don't know. Sure we can move it, but what would it be useful for? We'd also have to create a migration.
And that would be a block for only 4 settings.

As you wish.

Reply:
Mainly incase we need to expand. And it'd make more sense for consistency sake I meant. But I'm open to suggestions

Reply:
My main argument against it is the need of yet another migration script, even if it can perfectly be done. IMO the less migration the better.
But yes it would be more consistent to have it separated.